src/gui/transferlistsortmodel.cpp

```diff
@@ -70,12 +61,27 @@ namespace
     }
 
     template <typename T>
-    int customCompare(const T left, const T right)
+    concept Validateable = requires (T t) { {t.isValid()} -> std::same_as<bool>; };
+
+    template <Validateable T>
+    bool isValid(const T &value)
+    {
+        return value.isValid();
+    }
+
+    // consider negative values as invalid
+    template <typename T>
+        requires std::is_arithmetic_v<T>
+    bool isValid(const T value)
     {
-        static_assert(std::is_arithmetic_v<T>);
+        return (value >= 0);
+    }
 
-        const bool isLeftValid = (left >= 0);
-        const bool isRightValid = (right >= 0);
+    template <typename T>
+    int customCompare(const T &left, const T &right)
+    {
+        const bool isLeftValid = isValid(left);
+        const bool isRightValid = isValid(right);
```

Comment:
@glassez 
This new comparison logic isn't equivalent to the removed `int customCompare(const QDateTime &left, const QDateTime &right)`.
The `if (!isLeftValid && !isRightValid)` would produce unexpected results.

Furthermore, this `customCompare` was really intended for specific arithmetic types and not general use.
I would suggest reverting changes to this file in #21041.

Reply:
> Furthermore, this `customCompare` was really intended for specific arithmetic types and not general use.

I don't know how you define it, but this compare logic is really the same for all types for which we define the concept of "invalid value". The only difference is how the concept of "invalid value" is defined for a particular type.

Reply:
> I would suggest reverting these changes in #21041.

Done anyway. I don't want to waste my time on this issue anymore, it's already taken up too much of it.

Reply:
>I don't know how you define it, but this compare logic is really the same for all types for which we define the concept of "invalid value". The only difference is how the concept of "invalid value" is defined for a particular type.

For me, I would treat the comparison for each column as unique and it should be (easily) customizable. If some of them could share the same code then it is great and can be done. Also generalization should not become the final goal as it would hide the salient comparison of the column values (and possibly introduce slower comparison code).